### PR TITLE
Add form to Blog App

### PR DESCRIPTION
**Pull request by zunairkhan811**
- Add current_user method in Application Controller
- Add post-creation form using current_user
- Add comment creation form
- Add Like form
- Fix linter errors

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,6 +9,15 @@ def new
...
 
+  def create
+    @post = @user.posts.new(post_params)
+    if @post.save
+      redirect_to user_posts_path(current_user), notice: 'Your Post has been successfully Created'
+    else
+      render :new, notice: error
+    end
+  end
+
```
Comment (Munsa1) [inline app/controllers/posts_controller.rb:20]: - [ ] Great job implementing the create actions, however, we need to be able to create a post for the currently logged-in user. Currently, all posts created are being made for the first user only, kindly fix this.

Video example
[cinnamon-2023-12-08T221430+0200.webm](https://github.com/zunairkhan811/blog-app/assets/81932478/6e9c99a2-644f-4dac-abc8-55e1552aa78e)

- [ ] After fixing this issue kindly ensure that the user will be able to add comments and likes as well to the currently logged-in user

Example code snippet


  class PostsController < ApplicationController

    ...

    def new
      @post = Post.new
    end

    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end

    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end

	  ...

  end

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.new(user_id: current_user.id, post_id: params[:post_id], text: params[:text])
+    if @comment.save
+      redirect_to user_post_path(:user_id, :post_id), notice: 'Your comment has been successfully created.'
+    else
+      redirect_to user_post_path(:user_id, :post_id), alert: 'Error creating comment.'
+    end
+  end
+end
```
Comment (Munsa1) [inline app/controllers/comments_controller.rb:10]: - [ ] Again here to create a comment for the currently logged-in user, we need to refactor our controller. The comments controller should have new and create actions similar to the one shown below:

Example
```

class CommentsController < ApplicationController
 

  def new
    @comment = Comment.new
  end

	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end

  private 
  def comment_params
    params.require(:comment).permit(:text)
  end


end

```

Comment (Munsa1) [inline app/controllers/comments_controller.rb:10]: - [ ] There should be a form in the views/comments/new.html.erb file that accepts input for creating comments as per requirement, kindly refactor your folder and file structure to meet this requirement.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new(user_id: params[:user_id], post_id: params[:post_id])
+    if @like.save
+      redirect_to redirect_url, notice: 'You have successfully liked the post'
+    else
+      redirect_to redirect_url, alert: 'Error occurred while liking the post'
+    end
+  end
+
```
Comment (Munsa1) [inline app/controllers/likes_controller.rb:10]: - [ ] (Optional) Kindly ensure that when you refactor your codebase, you are able to add likes to the currently logged-in user. Currently, this is the case, this is just a reminder to ensure this doesn't break with any refactor.

Example code using the current user to apply likes.
    ...

    def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
	  ...

  end

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @zunairkhan811, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Created Forms :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (zunairkhan811) [reply app/controllers/posts_controller.rb:20]: Hi! @Munsa1, I hope you would be fine Can you please specify to the line that is it a project requirement? 
As far I do understand, this line clearly specifies that current_user should be first user from database..

image

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (zunairkhan811) [reply app/controllers/comments_controller.rb:10]: As stated above, as it belongs to current_user method and not included in a project Requirement.

image

Comment (zunairkhan811) [reply app/controllers/comments_controller.rb:10]: As stated above, as it belongs to current_user method and not included in a project Requirement.

image

File: app/controllers/likes_controller.rb (hunk 3 above)
Comment (zunairkhan811) [reply app/controllers/likes_controller.rb:10]: As stated above, as it belongs to current_user method and not included in a project Requirement.

image

**Comment by topeogunleye** [comment]
@zunairkhan811, Hi, I am your next code reviewer, please kindly push your changes to GitHub.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (topeogunleye) [reply app/controllers/comments_controller.rb:10]: @zunairkhan811, Kindly take note that you still need to create a form for comments as seen in the project requirements:

image

**Review by topeogunleye — APPROVED** [review]
### Hi @zunairkhan811, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
1. **`ApplicationController` - `current_user` Method:**
   - The `current_user` method in the `ApplicationController` returns the first user, providing a basic implementation for tracking the current user.

2. **`PostsController` - New & Create Actions:**
   - The `PostsController` effectively implements the new and create actions.
   - The `create` action handles post creation, utilizing `@current_user` for association and responding appropriately based on success or failure.

3. **Routes for Posts:**
   - Routes for creating posts are correctly configured in `config/routes.rb`.

4. **Post Creation Form:**
   - The `views/posts/new.html.erb` file contains a form for accepting post creation input.

5. **`CommentsController` - New & Create Actions:**
   - The `CommentsController` appropriately handles the new and create actions.
   - Comment creation involves associating the comment with the current user and the corresponding post.

6. **Routes for Comments:**
   - Routes for creating comments are correctly configured in `config/routes.rb`.

7. **Comment Creation Form:**
   - The `views/comments/new.html.erb` file contains a form for accepting comment creation input.

8. **`LikesController` - Create Action:**
   - The `LikesController` has an effective create action for adding likes to posts.

9. **Route for Likes:**
   - The route for creating likes is correctly configured in `config/routes.rb`.
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**